mongodb/helper/execode.py

```diff
@@ -89,46 +112,51 @@ def _excde_user_data_integration(action_model: ExecodeEntryModel, xparams: dict)
 class ExecodeParameterCollator:
     @staticmethod
     def collate_parameters(action: Execode, xparams: dict) -> dict:
+        ExecodeParameterCollator._check_missing_keys(action, xparams)
+
         if action == Execode.AR_ADD:
             return ExecodeParameterCollator._execode_ar_add(action, xparams)
-        elif action == Execode.REGISTER_CHANNEL:
+
+        if action == Execode.REGISTER_CHANNEL:
             return ExecodeParameterCollator._execode_conn_channel(xparams)
-        elif action == Execode.INTEGRATE_USER_DATA:
+
+        if action == Execode.INTEGRATE_USER_DATA:
             return ExecodeParameterCollator._execode_user_integrate(action, xparams)
-        else:
-            return xparams
+
+        return xparams
+
+    @staticmethod
+    def _check_missing_keys(action: Execode, xparams: dict):
+        required_keys = ExecodeRequiredKeys.get_required_keys(action)
+
+        missing_keys = required_keys.difference(xparams)
+        if missing_keys:
+            raise ExecodeCollationError(action, missing_keys, ExecodeCollationFailedReason.MISSING_KEY)
 
     @staticmethod
     def _execode_ar_add(action: Execode, xparams: dict) -> dict:
         k = param.AutoReply.PLATFORM
-        if xparams.get(k) is None or len(xparams[k][0]) == 0:
+        if isinstance(xparams[k], str) and len(xparams[k]) == 0:
             raise ExecodeCollationError(action, k, ExecodeCollationFailedReason.EMPTY_CONTENT)
-        else:
-            try:
-                xparams[k] = Platform.cast(xparams[k][0])
-            except KeyError as e:
-                raise ExecodeCollationError(action, k, ExecodeCollationFailedReason.MISSING_KEY, e)
-            except Exception as e:
-                raise ExecodeCollationError(action, k, ExecodeCollationFailedReason.MISC, e)
 
-        k = param.AutoReply.CHANNEL_TOKEN
-        xparams[k] = xparams[k][0]
+        try:
+            xparams[k] = Platform.cast(xparams[k])
+        except KeyError as e:
+            raise ExecodeCollationError(action, k, ExecodeCollationFailedReason.MISSING_KEY, e)
```

Comment:
Codacy found an issue: [Consider explicitly re-raising using the 'from' keyword](https://app.codacy.com/manual/RaenonX/Jelly-Bot/pullRequest?prid=6055131)